server/api/routes/auth.js

```diff
@@ -40,37 +40,47 @@ router.get('/callback', (req, res, next) => {
       return res.redirect('/login');
     }
 
-    req.logIn(user, (error) => {
+    req.logIn(user, async (error) => {
       if (error) {
         return next(error);
       }
       
-      // store user_id?
       const userInfo = {
         name: user.displayName,
         identifier: user.id
       };
       
-      // User.createUser(userInfo);
+      // After retrieving user from db, redirect them to their user page.
+      // Try to find a user with their user.id in the users db.
+      let userExists;
+      let data;
+      
       try {
-        users.findUser(user.id).then(data => {
-          if (data.length === 0) {
-            try {
-              users.createUser(userInfo);
-            } catch (e) {
-              console.log('Error Creating User: ', e);
-            }
-          }
-        });
-      } catch (e) {
+        userExists = await users.getUserByIdentifier(userInfo.identifier).then(user => data = user);
+      } catch(e) {
         console.log('Error finding user: ', e);
       }
-  
-      const { returnTo } = req.session;
+
+      if (!userExists) {
+        // Try to create a user.
+        try {
+          data = await users.createUser(userInfo);
+        } catch(e) {
+          console.log('Error creating user: ', e);
+        }
+      }
+
+      const { user_uuid } = data;
+      // store user's id from db in session storage for quering subscriptions
+      req.session.userInfo = {
+        user_id: data.id
+      };
```

Comment:
can also destructure the `id`

```suggestion
      const { user_uuid, id } = data;
      // store user's id from db in session storage for quering subscriptions
      req.session.userInfo = {
        user_id: id
      };
```